client/packages/system/src/ContactTrace/DetailView/DetailView.tsx

```diff
@@ -0,0 +1,270 @@
+import React, { FC, useEffect, useState } from 'react';
+import {
+  useTranslation,
+  DetailViewSkeleton,
+  AlertModal,
+  useNavigate,
+  RouteBuilder,
+  useDebounceCallback,
+  useBreadcrumbs,
+  useFormatDateTime,
+  Breadcrumb,
+  useIntlUtils,
+  ContactTraceNodeStatus,
+} from '@openmsupply-client/common';
+import {
+  useJsonForms,
+  useContactTraces,
+  ContactTraceRowFragment,
+  useDocumentRegistry,
+  SchemaData,
+} from '@openmsupply-client/programs';
+import { AppRoute } from '@openmsupply-client/config';
+import { Toolbar } from './Toolbar';
+import { Footer } from './Footer';
+import { PatientTabValue } from '../../Patient/PatientView/PatientView';
+import { usePatient } from '../../Patient';
+
+export type ContactTrace = {
+  status: ContactTraceNodeStatus;
+  datetime: string;
+};
+
+export type ContactTraceData = {
+  id: string | undefined;
+  type: string;
+  documentName: string | undefined;
+  contactTrace: ContactTrace;
+  patient: {
+    id: string;
+    name?: string | null;
+    firstName?: string | null;
+    lastName?: string | null;
+  };
+  schema: SchemaData | undefined;
+  programName: string;
+};
+
+/**
+ * Hook to provide all data needed for the DetailView.
+ * The Data is either load from an existing contact trace otherwise some default data is returned.
+ */
+const useContactTraceData = (
+  traceId: string,
+  createType: string | null,
+  createPatientId: string | null
+): { data: ContactTraceData | undefined; isLoading: boolean } => {
+  const [result, setResult] = useState<{
+    data: ContactTraceData | undefined;
```

Comment:
Similar to previous comment, I want to enforce that a data field is returned in all code paths, e.g. so that I don't forget to return the data in some cases.